sssom/parsers.py

```diff
@@ -499,6 +522,159 @@ def from_obographs(
+     specified by the effectiveTime field.,
+    - moduleId,SctId,Identifies the member version’s module. Set to a child of 900000000000443000|Module| within the
+    metadata hierarchy.,The only value in the entire set is '5991000124107', which has label 'SNOMED CT to ICD-10-CM
+    rule-based mapping module' (
+    https://www.findacode.com/snomed/5991000124107--snomed-ct-to-icd-10-cm-rule-based-mapping-module.html).
+    - refSetId,SctId,Set to one of the children of the |Complex map type| concept in the metadata hierarchy.,The only
+    value in the entire set is '5991000124107', which has label 'ICD-10-CM complex map reference set' (
+    https://www.findacode.com/snomed/6011000124106--icd-10-cm-complex-map-reference-set.html).
+    - referencedComponentId,SctId,The SNOMED CT source concept ID that is the subject of the map record.,
+    - mapGroup,Integer,An integer identifying a grouping of complex map records which will designate one map target at
+    the time of map rule evaluation. Source concepts that require two map targets for classification will have two sets
+    of map groups.,
+    - mapPriority,Integer,Within a map group, the mapPriority specifies the order in which complex map records should be
+    evaluated to determine the correct map target.,
+    - mapRule,String,A machine-readable rule, (evaluating to either ‘true’ or ‘false’ at run-time) that indicates
+    whether this map record should be selected within its map group.,
+    - mapAdvice,String,Human-readable advice that may be employed by the software vendor to give an end-user advice on
+    selection of the appropriate target code. This includes a) a summary statement of the map rule logic, b) a statement
+    of any limitations of the map record and c) additional classification guidance for the coding professional.,
+    - mapTarget,String,The target ICD-10 classification code of the map record.,
+    - correlationId,SctId,A child of |Map correlation value| in the metadata hierarchy, identifying the correlation
+    between the SNOMED CT concept and the target code.,
+    - mapCategoryId,SctId,Identifies the SNOMED CT concept in the metadata hierarchy which is the MapCategory for the
+    associated map record. This is a subtype of 447634004 |ICD-10 Map Category value|.,
+    """
+    # Local variables
+    # https://www.findacode.com/snomed/447561005--snomed-ct-source-code-to-target-map-correlation-not-specified.html
+    match_type_snomed_unspecified_id = 447561005
+    # - Note: joeflack4: I used this info as a reference for this pattern.
+    # https://www.medicalbillingandcoding.org/icd-10-cm/#:~:text=ICD%2D10%2DCM%20is%20a,decimal%20point%20and%20the%20subcategory.
+    always_confidence_pattern = 'ALWAYS [A-Z]{1}[0-9]{1,2}\.[0-9A-Z]{1,4}'
+    always_confidence_antipattern = always_confidence_pattern + '\?'
+    prefix_map = _ensure_prefix_map(prefix_map)
+    ms = _init_mapping_set(meta)
+
+    # Filtering
+    if filter_by_confident_mappings:
```

Comment:
Here's the filtering you requested.

It turns out that I was wrong earlier when I thought that if `mapAdvice` contained `'ALWAYS <code>'`, `mapRule` would always be `TRUE`. If this was the case, filtering by `mapRule` would've been much easier, but unfortunately it did not turn out to be the case.

I was thinking about a simple `.str.contains('ALWAYS')`, but it is possible that the word 'ALWAYS' could appear in cases other than `'ALWAYS <code>'`, so I decided to go with the stricter approach of a regex.

I created my own regex. But now I just realized that ICD10 is such a big thing that I should've just been googled a pre-baked regex. Here's one I found: https://www.johndcook.com/blog/2019/05/05/regex_icd_codes/. If you'd like me to use that or another regex instead, let me know.

I checked to see how many rows were removed by this filter, and it was about ~50%.

Reply:
@matentzn